--- mtr-0.95/bash-completion/mtr.orig	2022-01-11 10:10:34.000000000 +0100
+++ mtr-0.95/bash-completion/mtr	2022-01-13 21:13:17.193803957 +0100
@@ -1,9 +1,8 @@
-_mtr_module()
+_mtr()
 {
-  local cur prev OPTS
-  COMPREPLY=()
-  cur="${COMP_WORDS[COMP_CWORD]}"
-  prev="${COMP_WORDS[COMP_CWORD-1]}"
+  local cur prev words cword;
+  _init_completion -n : || return;
+
   case $prev in
     '-F'|'--filename')
       local IFS=$'\n'
@@ -12,7 +11,8 @@
       return 0
       ;;
     '-a'|'--address')
-      COMPREPLY=( $(compgen -W "ADDRESS" -- "$cur") )
+      _ip_addresses
+      __ltrim_colon_completions "$cur"
       return 0
       ;;
     '-f'|'--first-ttl'|'-m'|'--max-ttl'|'-U'|'--max-unknown'|'-B'|'--bitpattern'|'-Q'|'--tos'|'-c'|'--report-cycles')
@@ -62,7 +62,7 @@
       return 0
       ;;
   esac
-  COMPREPLY=( $(compgen -W "ip_address hostname" -- "$cur") )
+  _known_hosts
   return 0
 }
-complete -F _mtr_module mtr
+complete -F _mtr mtr
