--- mtr-0.92/ui/mtr.c~	2017-06-02 10:54:58.000000000 +0200
+++ mtr-0.92/ui/mtr.c	2017-08-13 10:33:42.063498074 +0200
@@ -741,6 +741,9 @@ int main(
 
     parse_mtr_options(&ctl, &names_head, getenv("MTR_OPTIONS"));
 
+    if ((strlen(argv[0]) > 3) && !strcmp(argv[0] + strlen(argv[0]) - 4, "mtr6"))
+        ctl.af = AF_INET6;
+
     parse_arg(&ctl, &names_head, argc, argv);
 
     while (optind < argc) {
