--- mtr-0.71/configure.in.orig	2006-03-24 07:37:17.000000000 +0100
+++ mtr-0.71/configure.in	2006-03-25 16:00:03.264992250 +0100
@@ -85,10 +85,11 @@
 AC_CHECK_FUNCS(seteuid)
 #  AC_CHECK_FUNC(setuid, , AC_MSG_ERROR (I Need either seteuid or setuid))
 
-AC_CHECK_FUNC(res_mkquery, , 
-  AC_CHECK_LIB(bind, res_mkquery, , 
-   AC_CHECK_LIB(resolv, res_mkquery, ,
-     AC_CHECK_LIB(resolv, __res_mkquery, , AC_MSG_ERROR(No resolver library found)))))
+AC_CHECK_FUNC(res_mkquery, ,[
+ AC_CHECK_FUNC(__res_mkquery, ,[
+  AC_CHECK_LIB(resolv, res_mkquery, ,[
+   AC_CHECK_LIB(resolv, __res_mkquery, ,[
+     AC_CHECK_LIB(bind, res_mkquery, , [AC_MSG_ERROR(No resolver library found)])])])])])
 # This next line would override the just detected-or-not -lresolv. 
 # This apparently hurts BSD. And it's bad practise. So it should go. 
 # However, it probably didn't get added for nothing..... Holler if
