diff -u mtr-0.87.orig/Makefile.am mtr-0.87/Makefile.am
--- mtr-0.87.orig/Makefile.am	2016-08-03 07:27:03.000000000 +0200
+++ mtr-0.87/Makefile.am	2017-01-12 18:55:47.832246506 +0100
@@ -2,9 +2,6 @@
 
 sbin_PROGRAMS = mtr
 man_MANS = mtr.8
-install-exec-hook: 
-	`setcap cap_net_raw+ep $(DESTDIR)$(sbindir)/mtr` \
-	|| chmod u+s $(DESTDIR)$(sbindir)/mtr
 
 mtr_SOURCES = mtr.c \
               net.c net.h \
Common subdirectories: mtr-0.87.orig/img and mtr-0.87/img
