diff -urN mtr-0.69.org/mtr.c mtr-0.69/mtr.c
--- mtr-0.69.org/mtr.c	2005-01-11 09:33:41.000000000 +0100
+++ mtr-0.69/mtr.c	2005-07-23 21:51:44.562847800 +0200
@@ -387,12 +387,6 @@
     exit(1);
   }
 
-  
-  if (net_set_interfaceaddress (InterfaceAddress) != 0) {
-    fprintf (stderr, "mtr: Couldn't set interface addres.\n"); 
-    exit (1); 
-  }
-
 #ifdef ENABLE_IPV6
   /* gethostbyname2() is deprecated so we'll use getaddrinfo() instead. */
   bzero( &hints, sizeof hints );
@@ -442,6 +436,11 @@
         exit(1);
       }
 
+  if (net_set_interfaceaddress (InterfaceAddress) != 0) {
+    fprintf (stderr, "mtr: Couldn't set interface addres.\n"); 
+    exit (1); 
+  }
+
   display_open();
   dns_open();
 
