diff -uNr mtr-0.69.orig/mtr.c mtr-0.69/mtr.c
--- mtr-0.69.orig/mtr.c	2005-09-18 10:54:21.000000000 +0200
+++ mtr-0.69/mtr.c	2005-09-18 11:31:50.000000000 +0200
@@ -74,7 +74,7 @@
 
 
 /* default display field(defined by key in net.h) and order */
-unsigned char fld_active[2*MAXFLD] = "LS NABWV";
+unsigned char fld_active[2*MAXFLD] = "LSR NABWV";
 int           fld_index[256];
 char          available_options[MAXFLD];
 
