diff -Nur mtr-0.54.orig/net.c mtr-0.54/net.c
--- mtr-0.54.orig/net.c	2003-05-11 11:10:55.000000000 +0200
+++ mtr-0.54/net.c	2003-05-11 11:24:41.000000000 +0200
@@ -267,7 +267,8 @@
   }
   if (rv < 0) {
     perror("sendto");
-    exit(1);
+    fprintf(stderr,"\r");
+    return;
   }
   first = 0;
 }
@@ -294,7 +295,8 @@
   if (setsockopt(sendsock6, IPPROTO_IPV6, IPV6_UNICAST_HOPS,
 	&hops, sizeof(hops)) < 0) {
     perror("setsockopt(IPV6_UNICAST_HOPS)");
-    exit(1);
+    fprintf(stderr,"\r");
+    return;
   }
 
   icmp6 = (struct icmp6_hdr *)packet;
@@ -316,7 +318,8 @@
 	);
   if (rv < 0) {
     perror("sendto");
-    exit(1);
+    fprintf(stderr,"\r");
+    return;
   }
   first = 0;
 }
