--- mtr-0.92/ui/mtr.c~	2017-08-13 10:34:14.000000000 +0200
+++ mtr-0.92/ui/mtr.c	2017-08-13 10:34:43.735191142 +0200
@@ -715,7 +715,7 @@ int main(
     ctl.probe_timeout = 10 * 1000000;
     ctl.ipinfo_no = -1;
     ctl.ipinfo_max = -1;
-    xstrncpy(ctl.fld_active, "LS NABWV", 2 * MAXFLD);
+    xstrncpy(ctl.fld_active, "LSR NABWV", 2 * MAXFLD);
 
     /*
        mtr used to be suid root.  It should not be with this version.
