diff -ru mtr-0.41.old/configure.in mtr-0.33/configure.in
--- mtr-0.41.old/configure.in	Tue Mar  9 19:56:49 1999
+++ mtr-0.41/configure.in	Thu Mar 11 21:26:32 1999
@@ -45,6 +45,9 @@
   AC_CHECK_LIB(bind, res_init, , 
    AC_CHECK_LIB(resolv, res_init, , AC_MSG_ERROR(No resolver library found))))
 
+AC_CHECK_FUNC(dn_expand, , 
+   AC_CHECK_LIB(resolv, dn_expand, , AC_MSG_ERROR(No resolver library found)))
+
 AC_CHECK_FUNC(herror, , AC_DEFINE(NO_HERROR))
 AC_CHECK_FUNC(strerror, , AC_DEFINE(NO_STRERROR))
 
